### PR TITLE
Adds supply chain output details to workload get response

### LINKED ISSUE #248: Add StampedRef Resource Kind and Name in workload get output
### Description of problem

Currently, we show Supply chain Resources, Ready, Healthy and Time columns in workload get output. It would be great to see an additional column that shows which resource was stamped out by the supply chain.

Screen Shot 2022-08-02 at 5 48 58 PM

### Proposed solution

Add a column before Ready column that shows the Kind and Name of the resource stamped out by the supply chain.

### Example

```
   RESOURCE          OUTPUT                              READY   HEALTHY   TIME
   source-provider   ImageRepository/reg-test5           True    True      4d
   deliverable       Deliverable/reg-test5               True    True      4d
   image-builder     Image/reg-test5                     True    True      4d
   config-provider   PodIntent/reg-test5                 True    True      4d
   app-config        ConfigMap/reg-test5                 True    True      4d
   config-writer     Runnable/reg-test5-config-writer    True    True      4d
```

cc @danfein for design input related to column name

### PR DESCRIPTION
**Pull request by shaheerkootteeri**
Pull request

### What this PR does / why we need it
PR will add Supply chain output details to workload get command response.

### Which issue(s) this PR fixes

Fixes #248

### Describe testing done for PR

Added additional unit testing for supply chain details as well as workload get command.
installed the changes and tested workload get in local cluster

### Additional information or special notes for your reviewer

Signed-off-by: Shaheer Kootteeri [kshaheer@vmware.com](mailto:kshaheer@vmware.com)

### CODE REVIEW

**Comment by shaheerkootteeri** [comment]
image

**Comment by shaheerkootteeri** [comment]
image

**Review by shashwathi — COMMENTED** [review]
Pr could be lot more simple
- remove NotFound func 
-  In findOutputRef func remove all the else conditions and include 1 return statement

File: pkg/printer/workload_status_printer.go [hunk 1]
```diff
@@ -39,13 +39,14 @@ func WorkloadResourcesPrinter(w io.Writer, workload *cartov1alpha1.Workload) err
...
 		ready, elapsedTransitionTime := findConditionReady(resource.Conditions, cartov1alpha1.ConditionResourceReady)
-
+		outputRef := getOutputRef(resource)
 		row := metav1beta1.TableRow{
 			Cells: []interface{}{
 				resource.Name,
 				ready,
 				healthy,
 				elapsedTransitionTime,
+				outputRef,
```
Comment (shashwathi) [inline pkg/printer/workload_status_printer.go:49]: No need for var `outputRef` .

File: pkg/printer/workload_status_printer_test.go [hunk 2]
```diff
@@ -22,6 +22,7 @@ import (
 	"testing"
 
 	"github.com/google/go-cmp/cmp"
+	v1 "k8s.io/api/core/v1"
```
Comment (shashwathi) [inline pkg/printer/workload_status_printer_test.go:25]: Update to `corev1 "k8s.io/api/core/v1" `

File: pkg/printer/workload_status_printer_test.go [hunk 3]
```diff
@@ -221,10 +222,123 @@ func TestWorkloadResourcesPrinter(t *testing.T) {
...
+							Type:   cartov1alpha1.ConditionResourceSubmitted,
+							Status: metav1.ConditionTrue,
+						},
+						{
+							Type:   cartov1alpha1.ConditionResourceReady,
+							Status: metav1.ConditionTrue,
+						},
+					},
+					StampedRef: &v1.ObjectReference{
+
```
Comment (shashwathi) [inline pkg/printer/workload_status_printer_test.go:251]: remove extra line

File: pkg/printer/workload_status_printer_test.go [hunk 4]
```diff
@@ -221,10 +222,123 @@ func TestWorkloadResourcesPrinter(t *testing.T) {
...
+							Type:   cartov1alpha1.ConditionResourceSubmitted,
+							Status: metav1.ConditionFalse,
+						},
+						{
+							Type:   cartov1alpha1.ConditionResourceHealthy,
+							Status: metav1.ConditionFalse,
+						},
+					},
+					StampedRef: &v1.ObjectReference{
+						Kind: "",
```
Comment (shashwathi) [inline pkg/printer/workload_status_printer_test.go:289]: by default `Kind` and `Name` field will be empty so delete the fields.

**Review by shashwathi — COMMENTED** [review]
looks good.

**Review by warango4 — APPROVED** [review]

**Review by shashwathi — APPROVED** [review]

**Pull request merged**